lib/views/go_to_page.dart

```diff
@@ -148,13 +164,76 @@ class _GoToPageState extends State<GoToPage> {
                       },
                       onSuggestionSelected: (suggestion) {
                         this._destinTypeAheadController.text = suggestion;
-                        originDestination.add("$suggestion");
-                        widget.route(originDestination);
+                        FocusScope.of(context).requestFocus(
+                            destinationTextField); // to change the focus of the textfield to the second textfield
+                        for (RoomCoordinate room in roomsList) {
+                          if (room.roomId == suggestion) {
+                            originDestinationCoords[1] = room;
+                          }
+                        }
                       },
                     ),
                   ),
                 ],
               ),
+              SizedBox(height: 20.0),
+              Center(
+                child: FloatingActionButton(
+                  child: Icon(
+                    Icons.navigation,
+                    color: Colors.white,
+                    size: 40.0,
+                  ),
+                  backgroundColor: Color.fromRGBO(147, 0, 47, 1),
+                  elevation: 2,
+                  onPressed: () {
+                    print("THIS LISTTT IS longgg" + originDestinationCoords.length.toString());
+                    if (originDestinationCoords[0] != null && originDestinationCoords[1] != null && originDestinationCoords[0] != originDestinationCoords[1]) {
+                      Navigator.pop(context);
+                      widget.coordinates(originDestinationCoords);
+                    }
```

Comment:
Is this if statement necessary? Should the UI take this responsibility? The path finding algorithm in the back-end has no issues with being given a same start and end point. However, if this enhances user experience leave it. This is more of a question for critical thinking rather than a change request

Reply:
This if statement is needed to make sure that if the user is not setting up the origin and destination, the go button would not work for him.

Reply:
But it was also checking that start and end are not equal

Reply:
But I suppose stop the user in the front end is better than sending the user the trivial 0 direction, so it's all good 👍